src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1205,7 +1258,10 @@ void PolicyHandler::OnAllowSDLFunctionalityNotification(
   // Case, when specific device was changed
   connection_handler::DeviceHandle device_handle = 0u;
   if (device_specific) {
-    policy_manager_->SetUserConsentForDevice(device_mac, is_allowed);
+    {
+      sync_primitives::AutoLock lock(policy_manager_lock_);
+      policy_manager_->SetUserConsentForDevice(device_mac, is_allowed);
```

Comment:
@SNiukalov I see a lot of calls like policy_manager-> something + acquire/release lock
As we have C++11 support in open source, I propose you to declare a template like:
```
  template <typename Func, typename... Args>
  void CallPolicyManagerFunction(Func func, Args... args) {
    sync_primitives::AutoLock lock(policy_manager_lock_);
    ((*policy_manager_).*func)(args...);
  }
```
And simplify all these calls to a single line:
```
CallPolicyManagerFunction(&PolicyManager::RemovePendingPermissionChanges, policy_app_id);
```
Please analyze all such places and try this template instead

Reply:
@AKalinich-Luxoft 
 Fixed in: https://github.com/smartdevicelink/sdl_core/pull/2614/commits/793f178f719ad695595d1c8ad0ae40bcf265c9b5